node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -398,6 +449,37 @@ func (o OsDeviceConnectivityHelperGeneric) GetHostsIdByArrayIdentifier(arrayIden
 
 }
 
+func (o OsDeviceConnectivityHelperGeneric) IsMpathMatchVolumeId(dmPath string,
+	volumeUuidLower string, volumeNguid string) (bool, error) {
+	sgInqWwn, err := o.getSgInqWwnFromDmPath(dmPath)
+	if err != nil {
+		return false, err
+	}
+	if sgInqWwn == volumeUuidLower || sgInqWwn == volumeNguid {
+		return true, nil
+	}
+	return false, &ErrorWrongDeviceFound{dmPath, volumeUuidLower, sgInqWwn}
+}
+
+func (o OsDeviceConnectivityHelperGeneric) IsMpathMatchVolumeIdWithoutErrors(dmPath string, wnatedVolumeIds []string) bool {
+	sgInqWwn, _ := o.getSgInqWwnFromDmPath(dmPath)
+	for _, wnatedVolumeId := range wnatedVolumeIds {
+		if sgInqWwn == wnatedVolumeId {
+			return true
+		}
+	}
+	return false
+}
+
+func (o OsDeviceConnectivityHelperGeneric) getSgInqWwnFromDmPath(dmPath string) (string, error) {
```

Comment:
```suggestion
func (o OsDeviceConnectivityHelperGeneric) getSgInqWwn(dmPath string) (string, error) {
```
since we don't have other (non-`dmPath`) `getSgInqWwn` functions, I think it is clear from the parameter name

Reply:
actually, this method seems to only lower the wwn. can't it can be done in `GetWwnByScsiInq`?

Reply:
another option is to have a separate comparison `isSameId` function that lowers and compares its parameters, so that we won't have to lower anywhere else in the code

Reply:
done, removed `getSgInqWwnFromDmPath` function, moved the lower to `GetWwnByScsiInq` and added the `isSameId` function

Reply:
I meant that `isSameId` would do the lowering, so we won't have to lower [anywhere](https://github.com/IBM/ibm-block-csi-driver/blob/f9f084833062cb7a610690393db475274311ae0c/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L549) [else](https://github.com/IBM/ibm-block-csi-driver/blob/f9f084833062cb7a610690393db475274311ae0c/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L224) in the code

Reply:
I agree that I can move [these](https://github.com/IBM/ibm-block-csi-driver/blob/f9f084833062cb7a610690393db475274311ae0c/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L549) lower to the `isSameId` function, but I can't see how I can move the lower from [`GetVolumeIdVariations`](https://github.com/IBM/ibm-block-csi-driver/blob/f9f084833062cb7a610690393db475274311ae0c/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L224) function to `isSameId`, what do you think?

Reply:
good point. then let's move only these for which we agree

Reply:
done